src/DCMTrajectoryGeneratorHelper.cpp

```diff
@@ -435,82 +435,81 @@ bool DCMTrajectoryGeneratorHelper::computeFeetWeight(const std::vector<StepPhase
+            while (instant < endOfPhase){
+                m_weightInLeft[instant] = 0.0;
+                m_weightInRight[instant] = 1.0;
+                instant++;
+            }
 
-                ++leftState;
-            } else {
+            if (leftState + 1 == leftSteps.cend()){
+                std::cerr << "[DCMTrajectoryGenerator::computeFeetWeight] Something went wrong. The step phases are not coherent with the left foot." << std::endl; //It's not possible to have a swing phase as last phase.
+                return false;
+            }
 
-                leftYawAngle = leftState->angle;
-                leftFootPosition(0) = leftState->position(0) + cos(leftYawAngle) * m_leftZMPDelta(0) - sin(leftYawAngle) * m_leftZMPDelta(1);
-                leftFootPosition(1) = leftState->position(1) + sin(leftYawAngle) * m_leftZMPDelta(0) + cos(leftYawAngle) * m_leftZMPDelta(1);
+            ++leftState;
+        } else {
 
-                rightYawAngle = rightState->angle;
-                rightFootPosition(0) = rightState->position(0) + cos(rightYawAngle) * m_rightZMPDelta(0) - sin(rightYawAngle) * m_rightZMPDelta(1);
-                rightFootPosition(1) = rightState->position(1) + sin(rightYawAngle) * m_rightZMPDelta(0) + cos(rightYawAngle) * m_rightZMPDelta(1);
+            leftYawAngle = leftState->angle;
+            leftFootPosition(0) = leftState->position(0) + cos(leftYawAngle) * m_leftZMPDelta(0) - sin(leftYawAngle) * m_leftZMPDelta(1);
+            leftFootPosition(1) = leftState->position(1) + sin(leftYawAngle) * m_leftZMPDelta(0) + cos(leftYawAngle) * m_leftZMPDelta(1);
 
-                while (instant < endOfPhase){
-                    feetDistance = iDynTree::toEigen(rightFootPosition) - iDynTree::toEigen(leftFootPosition);
+            rightYawAngle = rightState->angle;
+            rightFootPosition(0) = rightState->position(0) + cos(rightYawAngle) * m_rightZMPDelta(0) - sin(rightYawAngle) * m_rightZMPDelta(1);
+            rightFootPosition(1) = rightState->position(1) + sin(rightYawAngle) * m_rightZMPDelta(0) + cos(rightYawAngle) * m_rightZMPDelta(1);
 
-                    ZMPDistanceFromLeftFoot = iDynTree::toEigen(zmpPosition[instant]) - iDynTree::toEigen(leftFootPosition);
+            while (instant < endOfPhase){
+                feetDistance = iDynTree::toEigen(rightFootPosition) - iDynTree::toEigen(leftFootPosition);
 
-                    m_weightInLeft[instant] = ZMPDistanceFromLeftFoot.norm() / feetDistance.norm();
-                    m_weightInRight[instant] = 1 - ZMPDistanceFromLeftFoot.norm() / feetDistance.norm();
-                    ++instant;
-                }
+                ZMPDistanceFromLeftFoot = iDynTree::toEigen(zmpPosition[instant]) - iDynTree::toEigen(leftFootPosition);
 
+                m_weightInLeft[instant] = std::abs(1 - ZMPDistanceFromLeftFoot.norm() / feetDistance.norm());
+                m_weightInRight[instant] = std::abs(ZMPDistanceFromLeftFoot.norm() / feetDistance.norm());
```

Comment:
```suggestion
                m_weightInLeft[instant] = std::min(1, std::max(0.0, 1 - ZMPDistanceFromLeftFoot.norm() / feetDistance.norm())));
                m_weightInRight[instant] = 1.0 - m_weightInLeft[instant];
```

To make sure that all the values are between 0 and 1 and they sum up to 1.

Reply:
Applied in d06e9843e77f097e6cda147e7ee15f1b727b620b